app/api/controllers/validators/resolvers/Resolver.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api.controllers.validators.resolvers
+
+import api.models.errors.{ InternalError, MtdError }
+import cats.data.Validated
+import cats.data.Validated.Valid
+
+/** Parses a raw value (e.g. String or JsValue) to a target type, validating in the process.
+  *
+  * @tparam S
+  *   The source data type
+  * @tparam T
+  *   The target data type
+  */
+trait Resolver[S, T] {
```

Comment:
Is it not worth having a test file for this?

Reply:
sure, will add one